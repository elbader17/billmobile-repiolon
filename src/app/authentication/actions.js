import { Auth } from 'aws-amplify';
import { SET_JWT_TOKEN } from './constants';


const signIn = function(email, password) {
  if(email.length == 0 || password.length == 0){
    return {};
  }
  return (dispatch) => {
    const a = Auth.signIn(email, password)
    .then((data) => {
      const { jwtToken } = data.signInUserSession.idToken;
      dispatch(SET_JWT_TOKEN, { jwtToken });
      console.error(jwtToken);
    });
    return Promise.resolve(
      a
    );
  }
}

export { signIn };