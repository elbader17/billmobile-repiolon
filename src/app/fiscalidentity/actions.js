
import axios from 'axios';

import {
  SET_FISCAL_IDENTITY,
} from './constants';

function setIdentitiFiscal(name, cuit) {
  return {
    type: SET_FISCAL_IDENTITY,
    name,
    cuit,
  };
}

// eslint-disable-next-line func-names
const registerFiscalIdentity = function (name, cuit, jwtToken) {

  const instance = axios.create({
    baseURL: 'http://192.168.1.18:8888/',
    timeout: 1000,
    headers: { 'JWT-TOKEN': jwtToken },
  });

  const resource = {
    category: 'monotributo',
    name,
    identification: cuit,
  };

  return (dispatch) => {
    return instance.put('/v1/my....', { resource })
      .then(() => {
        dispatch(setIdentitiFiscal(name, cuit));
      })
      .catch((error) => {
        console.log(error);
      });
  };
};

export { registerFiscalIdentity };
