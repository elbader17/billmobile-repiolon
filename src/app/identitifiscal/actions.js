
import axios from 'axios';

import {
  SET_IDENTITI_FISCAL,
} from './constants';

function setIdentitiFiscal(name, cuit) {
  return {
    type: SET_IDENTITI_FISCAL,
    name,
    cuit,
  };
}

// eslint-disable-next-line func-names
const registerFiscalIdentiti = function (name, cuit, jwtToken) {

  const instance = axios.create({
    baseURL: 'http://192.168.1.18:8888/',
    timeout: 1000,
    headers: { 'JWT-TOKEN': jwtToken },
  });

  const resource = {
    category:'monotributo',
    name:name,
    identification:cuit,
  };

  return (dispatch) => {

    return instance.put('/v1/my....', {resource} )
      .then((response) => {
        dispatch(setIdentitiFiscal(name, cuit));
      })
      .catch((error) => {
        console.log(error);
      });
  };
  
};

export { registerFiscalIdentiti };

