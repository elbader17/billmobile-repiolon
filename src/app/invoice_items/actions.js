import { fetch_api } from '../../utils/fetchrefresh';
import {
  CREATE_INVOICE_ITEM,
  UPDATE_INVOICE_ITEM,
} from './constants';
import {
  createInvoice,
  getInvoice,
} from '../invoices/actions';

function createInvoiceItemAction(invoiceItem) {
  return {
    type: CREATE_INVOICE_ITEM,
    invoiceItem,
  };
}

function updateInvoiceItemAction(invoiceItem) {
  return {
    type: UPDATE_INVOICE_ITEM,
    invoiceItem,
  };
}

const updateInvoiceItem = (id, values) => {
  return (dispatch, getState) => {
    const { id: invoiceId } = getState().invoices.currentInvoice;
    return fetch_api(`/v1/invoice_items/${id}`, 'POST', false, 
    { resource: { ...values, invoice_id: invoiceId } })
        .then((response) => {
           console.log(response);
           dispatch(updateInvoiceItemAction(response.data));
           return dispatch(getInvoice(invoiceId));
        })
        .catch((error) => {
          console.log(error)
        });
  };
};

const deleteInvoiceItem = (id) => {
  return () => {
    return fetch_api(`/v1/invoice_items/${id}`,'GET', false)
      .then(() => console.log('OK'))
      .catch((error) => console.log(error));
  };
};

const getInvoiceItems = () => {
  return (getState) => {
    const { id } = getState().invoices.currentInvoice;
    return fetch_api(`/v1/invoice_items/${id}`,'GET', false)
      .then((response) => console.log(response))
      .catch((error) => console.log(error));
  };
};

const createInvoiceItem = ({category, name, price, quantity}) => {
  return (dispatch, getState) => {
    const { id: invoiceId } = getState().invoices.currentInvoice;
    let promise;
    if (invoiceId != null) {
      promise = Promise.resolve();
    } else {
      const { invoiceDate, voucherType } = getState().invoices.currentInvoice;
      promise = dispatch(createInvoice(invoiceDate, voucherType));
    }
    return promise.then(() => {
      const { id: updatedInvoiceId } = getState().invoices.currentInvoice;
      const resource = {
        category,
        name,
        price: parseFloat(price, 10),
        invoice_id: updatedInvoiceId,
        quantity
      };
      return fetch_api('/v1/invoice_items', 'POST', false, { resource })
        .then((response) => {
          dispatch(createInvoiceItemAction(response.data));
          return dispatch(getInvoice(updatedInvoiceId));
        })
        .catch((error) => {
          console.log(error.response);
        });
    });
  };
};

export { createInvoiceItem, updateInvoiceItem, deleteInvoiceItem, getInvoiceItems };
