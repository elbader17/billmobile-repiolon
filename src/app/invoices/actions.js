import { fetch_api } from '../../utils/fetchrefresh';
import {
  CREATE_INVOICE,
  UPDATE_INVOICE,
  LIST_INVOICE,
  GET_INVOICE,
} from './constants';

function createInvoiceAction(invoice) {
  return {
    type: CREATE_INVOICE,
    invoice,
  };
}

function updateInvoiceAction(invoice) {
  return {
    type: UPDATE_INVOICE,
    invoice,
  };
}

function getInvoiceAction(invoice) {
  return {
    type: GET_INVOICE,
    invoice,
  };
}

function listInvoiceAction(invoices) {
  return {
    type: LIST_INVOICE,
    invoices,
  };
}

function confirmInvoiceAction(invoice) {
  return {
    type: CONFIRM_INVOICE,
    invoice,
  };
}

const createInvoice = (invoiceDate, voucherType) => {
  const resource = {
    invoice_date: invoiceDate,
    invoice_type: voucherType
  };
  return (dispatch) => {
    return fetch_api('/v1/invoices', 'POST', false, { resource })
      .then((response) => dispatch(createInvoiceAction(response.data)))
      .catch((error) => console.log(error.response));
  };
};

const updateInvoice = (values) => {
  let resource = {};
  if (values.invoiceDate != null) 
    resource.invoice_date = values.invoiceDate;
  if (values.voucherType != null) 
    resource.invoice_type = values.voucherType;

  return (dispatch, getState) => {
    const { id: invoiceId } = getState().invoices.currentInvoice;
    return fetch_api(`/v1/invoices/${invoiceId}`,'PUT', false, { resource })
      .then((response) => dispatch(updateInvoiceAction(response.data)))
      .catch((error) => console.log(error));
  };
};

const listInvoice = () => {
  return (dispatch) => {
    return fetch_api('/v1/invoices', 'GET', false)
      .then((response) => {
        //console.log(response);
        dispatch(listInvoiceAction(response.data));
      })
      .catch((error) => {
        console.log(error)
      });
  };
};

const getInvoice = (id) => {
  return (dispatch) => {
    return fetch_api(`/v1/invoices/${id}`, 'GET', false)
      .then((response) => {
        dispatch(getInvoiceAction(response.data));
      })
      .catch((error) => {
        console.log(error);
      });
  };
};

const confirmInvoice = (attributes) => {
  return (dispatch, getState) => {
    const { id: invoiceId } = getState().invoices.currentInvoice;
    const resource = {
      invoice_id: invoiceId,
      state: attributes.state,
      total: attributes.total
    }
    return fetch_api(`/v1/invoices/${invoiceId}/confirm`,'PUT', false, { resource })
      .then((response) => {
        console.log(response);
        dispatch(confirmInvoiceAction(response.data));
      })
      .catch((error) => {
        console.log(error);
      });
  };
};

export { listInvoice, getInvoice, createInvoice, updateInvoice, confirmInvoice };
