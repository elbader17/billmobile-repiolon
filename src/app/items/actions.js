import axios from 'axios';
import {
  CREATE_ITEM,
  UPDATE_ITEM,
  LIST_ITEMS,
} from './constants';

function createItemAction(item) {
  return {
    type: CREATE_ITEM,
    item //id,attributes
  };
}

function updateItemAction(id, name, price) {
  const item = {id, name, price};
  return {
    type: UPDATE_ITEM,
    item
  };
}

function itemListAction(items) {
  return {
    type: LIST_ITEMS,
    items,
  };
}

const createItem = ({ category, name, price }) => {
  const resource = {
    category,
    name,
    price,
  };

  return (dispatch, getState) => {
    const instance = axios.create({
      headers: { 'JWT-TOKEN': getState().authentication.jwtToken },
    });
    return instance.post('v1/items', { resource })
      .then((response) => {
        return dispatch(createItemAction(response.data.data));
      })
      .catch((error) => {
        console.log(error);
      });
  };
};

const listItems = () => {
  return (dispatch) => {
    return axios.get('v1/items')
      .then((resources) => {
        console.log(resources);
        dispatch(itemListAction(resources.data));
      })
      .catch((error) => {
        console.log(error);
      });
  };
};

const updateItem = ({id, category, name, price}) => {
  const resource = {
    name,
    price,
    category
  };

  return (dispatch, getState) => {
    const instance = axios.create({
      headers: { 'JWT-TOKEN': getState().authentication.jwtToken },
    });
    return instance.put(`v1/items/${id}`, { resource })
      .then((response) => {
        const item = response.data.data;
        dispatch(updateItemAction(item.id, item.attributes.name, item.attributes.price));
      })
      .catch((error) => {
        console.log(error);
      });
  };
};

export { createItem, updateItem, listItems };
