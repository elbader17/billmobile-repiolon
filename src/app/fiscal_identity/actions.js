import axios from 'axios';
import { createInvoice } from '../invoices/actions';
import {
  ADD_FISCAL_IDENTIY_TO_INVOICE,
} from './constants';

const addfiscalIdentityToInvoiceAction = fiscalIdentity => {
  return {
    type: ADD_FISCAL_IDENTIY_TO_INVOICE,
    fiscalIdentity,
  };
}

const createFiscalIdentity = (resource, dispatch, getState) => {
  const instance = axios.create({
    headers: { 'JWT-TOKEN': getState().authentication.jwtToken },
  });
  return instance.post('/v1/invoices_fiscal_identities', { resource })
    .then((response) => {
      const { data } = response.data;
      dispatch(addfiscalIdentityToInvoiceAction(data));
    })
    .catch((error) => {
      console.log(error.response);
    });
};

const updateFiscalIdentity = (resource, dispatch, getState) => {
  const instance = axios.create({
    headers: { 'JWT-TOKEN': getState().authentication.jwtToken },
  });
  return instance.put(`/v1/invoices_fiscal_identities/${resource.id}`, { resource })
    .then((response) => {
      const { data } = response.data;
      dispatch(addfiscalIdentityToInvoiceAction(data));
    })
    .catch((error) => {
      console.log(error.response);
    });
};

const addFiscalIdentityToInvoice = (name, identity, id) => {
  return (dispatch, getState) => {
    const { id: invoiceId } = getState().invoices.currentInvoice;
    let promise;
    if (invoiceId != null) {
      promise = Promise.resolve();
    } else {
      const { invoiceDate, voucherType } = getState().invoices.currentInvoice;
      promise = dispatch(createInvoice(invoiceDate, voucherType));
    }
    return promise.then(() => {
      const updatedInvoiceId = getState().invoices.currentInvoice.id;
      const resource = {
        category: 'monotributo',
        name,
        id,
        identification: identity,
        invoice_id: updatedInvoiceId,
      };
      return createFiscalIdentity(resource, dispatch, getState);
    });
  };
};

export { addFiscalIdentityToInvoice };
