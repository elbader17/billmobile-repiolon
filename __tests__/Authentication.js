/**
 * @format
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */

import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Authentication from '../src/components/Authentication/index';
import { signIn } from '../src/app/authentication/actions';



/**
 * 
it('matches the snapshot', () => {
  const snapshot = shallow(<Authentication/>);
  expect(wrapper.find)
});
 */



it('matches the snapshot', () => {
  const tree = renderer.create(<Authentication />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('return empty signIn', () => {
  expect( signIn('','')).toEqual({});
})




it('test buton submitSignIn', function() {
  const wrapper = shallow(<Authentication />);
  let refreshData = jest.spyOn(wrapper.instance(), "submitSignIn");
  expect(wrapper.find('#submitSignIn')).toHaveLength(1);
  wrapper.find("[testID='submitSignIn']").onPress();
  expect(refreshData).toHaveBeenCalledTimes(1);
});