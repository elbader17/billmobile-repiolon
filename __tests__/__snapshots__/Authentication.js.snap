// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#e3e3e3",
        "borderRadius": 3,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 5,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 5,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomLeftRadius": 3,
          "borderRightColor": "#bdc6cf",
          "borderRightWidth": 0,
          "borderTopLeftRadius": 3,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
        testID="buttonGroupItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 16.25,
                "fontWeight": "500",
              }
            }
            testID="buttonGroupItemText"
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
          >
            Sign Up
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderBottomRightRadius": 3,
          "borderLeftColor": "#bdc6cf",
          "borderLeftWidth": 1,
          "borderTopRightRadius": 3,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
        testID="buttonGroupItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#5e6977",
                "fontSize": 16.25,
                "fontWeight": "500",
              }
            }
            testID="buttonGroupItemText"
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
          >
            Sign In
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View>
    <View>
      <View
        style={
          Object {
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#86939e",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Name
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 15,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
                underlayColor="white"
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "black",
                        "fontSize": 24,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="iconIcon"
                >
                  ïƒ 
                </Text>
              </View>
            </View>
          </View>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Name"
            style={
              Object {
                "alignSelf": "center",
                "color": "black",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
      <View
        style={
          Object {
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#86939e",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Email
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 15,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
                underlayColor="white"
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "black",
                        "fontSize": 24,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="iconIcon"
                >
                  ïƒ 
                </Text>
              </View>
            </View>
          </View>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="my@email.com"
            style={
              Object {
                "alignSelf": "center",
                "color": "black",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
      <View
        style={
          Object {
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#86939e",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Password
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 15,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
                underlayColor="white"
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "black",
                        "fontSize": 24,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="iconIcon"
                >
                  ï€£
                </Text>
              </View>
            </View>
          </View>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="p@ssw0rd123"
            secureTextEntry={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "black",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
      <View
        style={
          Object {
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#86939e",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Confirm Password
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 15,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
                underlayColor="white"
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "black",
                        "fontSize": 24,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="iconIcon"
                >
                  ï€£
                </Text>
              </View>
            </View>
          </View>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="p@ssw0rd123"
            secureTextEntry={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "black",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
      <View
        style={
          Object {
            "borderRadius": 3,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#2089dc",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Submit
            </Text>
          </View>
        </View>
      </View>
      <Modal
        hardwareAccelerated={false}
        visible={false}
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "paddingHorizontal": 10,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#86939e",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Confirmation Code
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderBottomWidth": 1,
                  "borderColor": "#86939e",
                  "flexDirection": "row",
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 40,
                    "justifyContent": "center",
                    "marginLeft": 15,
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                    theme={
                      Object {
                        "colors": Object {
                          "disabled": "hsl(208, 8%, 90%)",
                          "divider": "#bcbbc1",
                          "error": "#ff190c",
                          "grey0": "#393e42",
                          "grey1": "#43484d",
                          "grey2": "#5e6977",
                          "grey3": "#86939e",
                          "grey4": "#bdc6cf",
                          "grey5": "#e1e8ee",
                          "greyOutline": "#bbb",
                          "platform": Object {
                            "android": Object {
                              "error": "#f44336",
                              "primary": "#2196f3",
                              "secondary": "#9C27B0",
                              "success": "#4caf50",
                              "warning": "#ffeb3b",
                            },
                            "ios": Object {
                              "error": "#ff3b30",
                              "primary": "#007aff",
                              "secondary": "#5856d6",
                              "success": "#4cd964",
                              "warning": "#ffcc00",
                            },
                          },
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#8F0CE8",
                          "success": "#52c41a",
                          "warning": "#faad14",
                        },
                      }
                    }
                    underlayColor="white"
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "black",
                            "fontSize": 24,
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="iconIcon"
                    >
                      ï€£
                    </Text>
                  </View>
                </View>
              </View>
              <TextInput
                allowFontScaling={true}
                onChangeText={[Function]}
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "black",
                    "flex": 1,
                    "fontSize": 18,
                    "minHeight": 40,
                  }
                }
                testID="RNE__Input__text-input"
                underlineColorAndroid="transparent"
              />
            </View>
          </View>
          <View
            style={
              Object {
                "borderRadius": 3,
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#2089dc",
                    "borderColor": "#2089dc",
                    "borderRadius": 3,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "padding": 8,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 18,
                      "paddingBottom": 1,
                      "paddingTop": 2,
                      "textAlign": "center",
                    }
                  }
                >
                  Submit
                </Text>
              </View>
            </View>
          </View>
        </View>
      </Modal>
    </View>
  </View>
</View>
`;
